src/views/dashboard/users/AllUsers.vue

```diff
@@ -70,67 +81,176 @@
             } else {
               this.users = response.sort((a, b) => (a.modified_at < b.modified_at) ? 1 : -1)
-              this.alert = { show: true, text: 'Users loaded.', type: 'success' }
+              this.snack = { show: true, text: 'Users loaded.', color: 'success' }
             }
           })
       },
       deleteUser (i, user) {
         console.log('Deleting user with id: ' + user.id)
-        API.user.delete(user.email)
-          .then(() => {
-            this.users.splice(i, 1)
-            this.alert = { show: true, text: 'User ' + user.id + ' was deleted succesfully.', type: 'success' }
+        API.user.delete(user.id)
+          .then((response) => {
+            if ('error' in response) {
+              console.log(response)
+            } else {
+              this.users.splice(i, 1)
+              this.snack = { show: true, text: 'User ' + user.id + ' was deleted succesfully.', color: 'success' }
+            }
           })
           .catch((error) => {
             console.log(error)
-            this.alert = { show: true, text: 'There was an error deleting the user ' + user.id + '.', type: 'error' }
+            this.snack = { show: true, text: 'There was an error deleting the user ' + user.id + '.', color: 'error' }
           })
       },
-      editUser (i, user) {
-        console.log('Editing user with id: ' + user.id)
+      editUser (payload) {
+        console.log('Editing user with id: ' + this.userEdit.id)
+        console.log(payload)
+        this.userEdit.name = payload.name.value
+        this.userEdit.email = payload.email.value
+        // meter correo, nombre y revisar qué más
+        // quitar admin y hacerlo con switch
+        // mostrar de otra manera el superadmin
+        this.showEditModal = false
+        console.log(this.userEdit)
+        API.user.put(
+          this.userEdit.id,
+          { name: this.userEdit.name, email: this.userEdit.email },
+          { 'Content-Type': 'application/json' },
+        )
+          .then((response) => {
+            if ('error' in response) {
+              console.log(response.error)
+              this.snack = { show: true, text: 'There was an error editing the user: ' + response.error + '.', color: 'error' }
+            } else {
+              this.snack = { show: true, text: 'User ' + this.userEdit.id + ' was edited succesfully.', color: 'success' }
+            }
+          })
+          .catch((error) => {
+            console.log(error)
+            this.snack = { show: true, text: 'There was an error editing the user ' + this.userEdit.id + '.', color: 'error' }
+          })
+      },
+      editAdmin (user) {
+        this.userEdit = user
+        console.log('Editing user admin with id: ' + this.userEdit.id)
+        if (user.admin !== true) {
+          this.userEdit.admin = false
+        } else {
+          this.userEdit.admin = true
+        }
```

Comment:
I believe that all this is not needed. You could just use `user` in the rest of the function without using `this.userEdit`. The row is modified already by the switch. You only need to send the user information to the endpoint at this point.

Reply:
Done.